src/fruity/lockdown.vala

```diff
@@ -75,7 +75,19 @@ namespace Frida.Fruity {
 			closed ();
 		}
 
-		public async IOStream start_service (string name, Cancellable? cancellable = null) throws LockdownError, IOError {
+		public async IOStream start_service (string name_with_options, Cancellable? cancellable = null) throws LockdownError, IOError {
+			bool tls_handshake_only = false;
+			string name = name_with_options;
+
+			var options_start = name_with_options.index_of_char('?');
+			if (options_start != -1) {
+				if (options_start < name_with_options.char_count() - 1) {

```

Comment:
`char_count()` returns the number of characters, but `index_of_char ()` is the index in bytes.